### PR TITLE
fix: respect symbolOpacity, gradientOpacity, symbolFillColor, and symbolStrokeColor

### PR DESCRIPTION
**Pull request by domoritz**
Fixes https://github.com/vega/vega-lite/issues/6719

### CODE REVIEW

File: examples/compiled/circle_binned.vg.json [hunk 1]
```diff
@@ -157,7 +157,6 @@
   "legends": [
     {
       "size": "size",
-      "gradientLength": {"signal": "clamp(height, 64, 200)"},
```
Comment (donghaoren) [inline examples/compiled/circle_binned.vg.json:160]: Could you explain why this is removed?

File: examples/compiled/concat_marginal_histograms.vg.json [hunk 2]
```diff
@@ -441,7 +441,6 @@
     {
       "fill": "color",
       "gradientLength": {"signal": "clamp(concat_1_height, 64, 200)"},
-      "symbolType": "square",
```
Comment (donghaoren) [inline examples/compiled/concat_marginal_histograms.vg.json:444]: Why is this removed?

File: src/compile/legend/parse.ts [hunk 3]
```diff
@@ -126,6 +125,13 @@ export function parseLegendForChannel(model: UnitModel, channel: NonPositionScal
   };
 
   for (const property of LEGEND_COMPONENT_PROPERTIES) {
+    if (
+      (legendType === 'gradient' && property.startsWith('symbol')) ||
+      (legendType === 'symbol' && property.startsWith('gradient'))
+    ) {
+      continue;
```
Comment (donghaoren) [inline src/compile/legend/parse.ts:132]: So we are skipping gradient properties for symbol legend and vice versa, and that's why we are seeing some symbolType and gradientLength properties being removed, right?

Comment (domoritz) [reply src/compile/legend/parse.ts:132]: Yes!

File: examples/compiled/circle_binned.vg.json (hunk 1 above)
Comment (domoritz) [reply examples/compiled/circle_binned.vg.json:160]: The legend is a symbol legend and doesn't use gradient length. 

You found the code below: https://github.com/vega/vega-lite/pull/6724/files#r453887311

File: examples/compiled/concat_marginal_histograms.vg.json (hunk 2 above)
Comment (domoritz) [reply examples/compiled/concat_marginal_histograms.vg.json:444]: Same as above

File: examples/compiled/point_angle_windvector.vg.json [hunk 4]
```diff
@@ -87,7 +87,6 @@
   "legends": [
     {
       "size": "size",
-      "gradientLength": {"signal": "clamp(height, 64, 200)"},
```
Comment (Saba9) [inline examples/compiled/point_angle_windvector.vg.json:90]: Do you know why the PNG image generated for this spec changed?

Comment (domoritz) [reply examples/compiled/point_angle_windvector.vg.json:90]: Because the png generation isn't stable. I've seen this in a few places. 

Just ignore it when the SVG doesn't change either.

Comment (domoritz) [reply examples/compiled/point_angle_windvector.vg.json:90]: It would be good to figure out how we can minimize the noise @kanitw
